File: quartzwood/models/entity.py
from typing import Optional
from sqlmodel import SQLModel, Field
from quartzwood.models.enums import EntityType

class Entity(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(unique=True)
    type: EntityType = EntityType.personal
    location: Optional[str] = None

File: Quartzwood.Server/Models/Entity.cs
using System.ComponentModel.DataAnnotations;

namespace Quartzwood.Server.Models;

public enum EntityType
{
    Personal,
    Club,
    Friend,
    Shop
}

public class Entity
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();
    
    [Required]
    public string Name { get; set; } = string.Empty;
    
    public EntityType Type { get; set; } = EntityType.Personal;
    
    public string? Location { get; set; }
    
    // Navigation
    public ICollection<Group> Groups { get; set; } = new List<Group>();
}